src/sinks/util/buffer/compression.rs

```diff
@@ -0,0 +1,144 @@
+use serde::{de, ser};
+use std::fmt;
+
+#[derive(Debug, Derivative, Copy, Clone, Eq, PartialEq)]
+#[derivative(Default)]
+pub enum Compression {
+    #[derivative(Default)]
+    None,
+    Gzip(usize),
+}
+
+impl Compression {
+    pub fn default_gzip() -> Compression {
+        Compression::Gzip(6)
+    }
```

Comment:
Consider marking functions like this as `const` to allow them to be used as initializers for static data.